compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IrUtils.kt

```diff
@@ -440,18 +440,29 @@ fun irConstructorCall(
     }
 
 fun irCall(
-    call: IrMemberAccessExpression,
+    call: IrFunctionAccessExpression,
     newFunction: IrFunction,
-    dispatchReceiverAsFirstArgument: Boolean = false,
-    firstArgumentAsDispatchReceiver: Boolean = false
+    dispatchReceiverAsArgument: Boolean = false,
+    argumentAsDispatchReceiver: Boolean = false,
+    extensionReceiverAsArgument: Boolean = dispatchReceiverAsArgument,
```

Comment:
These defaults look confusing. Any objections on changing it to `false` to make intent explicit?

Reply:
I don't have any objections to making it false by default, since that was essentially the original behavior anyway. I will make this change.

The reason I did this is that I _thought_ it is more typical that both corresponding arguments (e.g., `dispatchReceiverAsArgument` and `extensionReceiverAsArgument`) would have the same value when this function is used. For example, in JVM, when methods are moved from an interface into the "DefaultImpls" class as a static method, both dispatch receiver and extension receiver become parameters.

I don't know enough about InlineClassLowering to determine why it needs to keep the extension receiver as-is. AFAIK extension receivers become parameters eventually anyway. But I don't have any objections to just making it false by default.

Reply:
Great! These functions are originated in JS IR BE where they are used extensively to translate missing built-ins (Any, Int, etc.) member calls into runtime plain function calls. These are concrete functions with known signature, AFAIK without extension receivers.